### PR TITLE
feat(input-password-toggle): add ionic theme design

### PR DESCRIPTION
**Pull request by joselrio**
Issue number: ROU-4860

---------

## What is the current behavior?
Input Password Toggle does not have styles for the ionic theme.

## What is the new behavior?
- Added color style
- Set margins

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

| Figma Design | Implemented Input Password Toggle |
| --- | --- | 
| Figma Design | Implemented InputPasswordToggle | 

- [Basic Preview](https://ionic-framework-git-rou-4860-ionic1.vercel.app/src/components/input-password-toggle/test/basic?ionic:theme=ionic)

### CODE REVIEW

File: core/src/components/input-password-toggle/input-password-toggle.ionic.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@use "../../foundations/ionic.vars.scss" as tokens;
```
Comment (brandyscarney) [inline core/src/components/input-password-toggle/input-password-toggle.ionic.scss:1]:
```suggestion
@use "../../themes/ionic/ionic.globals.scss" as globals;
```

File: core/src/components/input-password-toggle/input-password-toggle.ionic.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+@import "./input-password-toggle.common.scss";
+
+// Ionic Input Toggle Password
+// --------------------------------------------------
+:host ion-button.button-has-icon-only {
+  --padding-end: inherit;
+  --color: #{tokens.$ionic-color-neutral-500};
+  --margin-right: calc(-1 * var(--padding-end));
+
+  margin: 0 var(--margin-right) 0 0;
```
Comment (brandyscarney) [inline core/src/components/input-password-toggle/input-password-toggle.ionic.scss:11]:
```suggestion
  --margin-end: calc(-1 * var(--padding-end));

  @include margin(0, var(--margin-end), 0, 0);
```

We should use `margin-end` to match RTL naming

File: core/src/components/input-password-toggle/input-password-toggle.tsx [hunk 3]
```diff
@@ -1,10 +1,8 @@
-import eyeSlashRegular from '@phosphor-icons/core/assets/regular/eye-slash.svg';
-import eyeRegular from '@phosphor-icons/core/assets/regular/eye.svg';
 import type { ComponentInterface } from '@stencil/core';
 import { Component, Element, Host, Prop, h, Watch } from '@stencil/core';
 import { printIonWarning } from '@utils/logging';
 import { createColorClasses } from '@utils/theme';
-import { eyeOff, eye } from 'ionicons/icons';
+import { eyeOff, eye, eyeOutline, eyeOffOutline } from 'ionicons/icons';
```
[the remark this answers is not in the document]
Comment (joselrio) [reply core/src/components/input-password-toggle/input-password-toggle.tsx]: Thanks for raising it... I forget to undo this 🤦

File: core/src/components/input-password-toggle/input-password-toggle.ionic.scss [hunk 4]
```diff
@@ -0,0 +1,12 @@
+@use "../../themes/ionic/ionic.globals.scss" as globals;
+@import "./input-password-toggle.common.scss";
```
Comment (brandyscarney) [inline core/src/components/input-password-toggle/input-password-toggle.ionic.scss:2]:
```suggestion
@use "./input-password-toggle.common.scss";
```

File: core/src/components/input-password-toggle/input-password-toggle.ionic.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+// Ionic Input Toggle Password
+// --------------------------------------------------
+:host ion-button.button-has-icon-only {
+  --padding-end: inherit;
+  --color: #{globals.$ionic-color-neutral-500};
+  --margin-end: calc(-1 * var(--padding-end));
+
+  @include margin(0, var(--margin-end), 0, 0);
```
Comment (brandyscarney) [inline core/src/components/input-password-toggle/input-password-toggle.ionic.scss:11]:
```suggestion
  @include globals.margin(0, var(--margin-end), 0, 0);
```

**Review by brandyscarney — APPROVED** [review]
Looks good! Just recommend changing to `@use`.

**Pull request merged**